### PR TITLE
Workmarcon

### PR DESCRIPTION
**Pull request by marconepo7**
final

### CODE REVIEW

File: homework1.iml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+      <sourceFolder url="file://$MODULE_DIR$/src/main/java" isTestSource="false" />
+      <sourceFolder url="file://$MODULE_DIR$/src/test/java" isTestSource="true" />
+      <excludeFolder url="file://$MODULE_DIR$/target" />
+    </content>
+    <orderEntry type="inheritedJdk" />
+    <orderEntry type="sourceFolder" forTests="false" />
+    <orderEntry type="library" name="Maven: junit:junit:4.13" level="project" />
+    <orderEntry type="library" name="Maven: org.hamcrest:hamcrest-core:1.3" level="project" />
+  </component>
+</module>
```
Comment (pdgetrf) [inline homework1.iml:16]: do not check in ".idea" folder and the iml file

File: src/main/java/Problem1.java [hunk 2]
```diff
@@ -3,6 +3,22 @@
...
+            int mid = (right + left) / 2;
+
+            if (target == data[mid]) {
+                return mid;
+            } else if (target > data[mid]) {
+                left = mid + 1;
+            } else
+                right = mid - 1;
+        }
         return -1; // Place holder
```
Comment (pdgetrf) [inline src/main/java/Problem1.java:21]: remove "// place holder" when homework is done

File: src/test/java/Problem2Test.java [hunk 3]
```diff
@@ -8,7 +8,31 @@
...
-        // add your tests here
+        int [][] inputs = {
+                {3, 2, 7, 4, 8},
+                {5, 4, 3, 2, 1},
+                {23, 63, 26, 87, 57},
+                {8, 4},
+                {7, 7, 7, 4, 4,},
+
+        };
+        int [][] answer = {
```
Comment (pdgetrf) [inline src/test/java/Problem2Test.java:19]: great test!

consider add a few more for edge cases. for example, empty array, null array, single-element array.

**Review by pdgetrf — APPROVED** [review]
LGTM
